app/uk/gov/hmrc/apidefinition/connector/AWSAPIPublisherConnector.scala

```diff
@@ -56,20 +56,24 @@ class AWSAPIPublisherConnector @Inject() (
   val apiKeyHeaderName               = "x-api-key"
   val headers: Seq[(String, String)] = Seq(CONTENT_TYPE -> JSON, apiKeyHeaderName -> awsApiKey)
 
-  def createOrUpdateAPI(apiName: String, awsSwaggerDetails: AWSSwaggerDetails)(hc: HeaderCarrier): Future[String] = {
-    implicit val headersWithoutAuthorization: HeaderCarrier = hc.copy(authorization = None)
-    http.PUT[AWSSwaggerDetails, Either[UpstreamErrorResponse, RequestId]](s"$serviceBaseUrl/$apiName", awsSwaggerDetails, headers) flatMap {
+  def createOrUpdateAPI(apiName: String, awsSwaggerDetails: AWSSwaggerDetails)(implicit hc: HeaderCarrier): Future[String] = {
+    val headersWithoutAuthorization: HeaderCarrier = hc.copy(authorization = None)
+    val wts = implicitly[Writes[AWSSwaggerDetails]]
+    val rds = implicitly[HttpReads[Either[UpstreamErrorResponse, RequestId]]]
+
+    http.PUT[AWSSwaggerDetails, Either[UpstreamErrorResponse, RequestId]](s"$serviceBaseUrl/$apiName", awsSwaggerDetails, headers)(wts, rds, headersWithoutAuthorization, ec) flatMap {
```

Comment:
You should be able to use
`...)(implicitly, implicitly, headersWithoutAuthorization, implicitly)`

and not need local vals typed out.